Update build from source and getting started docs

GregoryComer opened the pull request:
### Summary

Update Getting Started and Build from Source Docs:
 * Integrate Windows steps into the main flow with minor Windows-specific callouts.
 * Clarify top-level flow for building from source - add a table by use case.
 * Clarify building ET as a submodule vs standalone build.
 * Re-order, re-word, and clean up the content related to building from source.
 * Add info on NDK build for Android.

Tracked in https://github.com/pytorch/executorch/issues/14791 and https://github.com/pytorch/executorch/issues/14759.

cc @mergennachin @AlannaBurke @byjlw

GregoryComer committed:
Update build from source and getting started docs

mergennachin reviewed (approved):
I see that you removed but I recommend adding back a brief "Verify Your Build" section after the Building section that contains 

  - One-line command to export a simple model
  - One-line command to run it with the executor_runner

GregoryComer commented:
> I see that you removed but I recommend adding back a brief "Verify Your Build" section after the Building section that contains
> 
> * One-line command to export a simple model
> * One-line command to run it with the executor_runner

Sounds good. I'll re-add. The reason I removed it was partly because we don't actually build the executor_runner for most presets and it requires multiple CLI options to include. I'm a little conflicted on this, but maybe we should build by default?

I think it does make sense to have a quick verification step. I'll re-add.

GregoryComer commented:
> > I see that you removed but I recommend adding back a brief "Verify Your Build" section after the Building section that contains
> > 
> > * One-line command to export a simple model
> > * One-line command to run it with the executor_runner
> 
> Sounds good. I'll re-add. The reason I removed it was partly because we don't actually build the executor_runner for most presets and it requires multiple CLI options to include. I'm a little conflicted on this, but maybe we should build by default?
> 
> I think it does make sense to have a quick verification step. I'll re-add.

Added.
